src/utils/Simplify.cpp

```diff
@@ -0,0 +1,175 @@
+//Copyright (c) 2022 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#include <limits>
+#include <queue> //Priority queue to prioritise removing unimportant vertices.
+
+#include "Simplify.h"
+
+namespace cura
+{
+
+Simplify::Simplify(const coord_t max_resolution, const coord_t max_deviation, const coord_t max_area_deviation)
+    : max_resolution(max_resolution)
+    , max_deviation(max_deviation)
+    , max_area_deviation(max_area_deviation)
+{}
+
+Simplify::Simplify(const Settings& settings)
+    : max_resolution(settings.get<coord_t>("meshfix_maximum_resolution"))
+    , max_deviation(settings.get<coord_t>("meshfix_maximum_deviation"))
+    , max_area_deviation(settings.get<coord_t>("meshfix_maximum_extrusion_area_deviation"))
```

Comment:
This divides the `meshfix_maximum_extrusion_area_deviation` value by 1000. This value is already in um^2.